.pre-commit-config.yaml

```
@@ -4,3 +4,24 @@ repos:
     hooks:
       - id: black
         language_version: python3
+  - repo: https://github.com/gitleaks/gitleaks
+    rev: v8.16.3
+    hooks:
+      - id: gitleaks
+  - repo: https://github.com/jumanjihouse/pre-commit-hooks
+    rev: 3.0.0
+    hooks:
+      - id: shellcheck
+  - repo: https://github.com/pocc/pre-commit-hooks
+    rev: v1.3.5
+    hooks:
+      - id: cpplint
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+  - repo: https://github.com/pylint-dev/pylint
+    rev: v2.17.2
+    hooks:
+      - id: pylint
```

`pylint` is already part of our CI. I don't mind having it as part of the pre-commit hooks also. If we include `pylint` as a hook, then we should add `flake8` as well.

---

`pylint` is quite problematic in pre-commit. Pre-commit runs non-local repos in a separate environment, where pylint dynamic code import fails. Required dependencies are however installed into the local dev environment, so the `local` hook could be used.

```yaml
- repo: local
  hooks:
    - id: pylint
      name: pylint
      entry: pylint
      language: system
      types: [python]
      require_serial: true
```
This may have other implications, so I'd suggest against this hook. However, other python hooks that work completely on the static level do work nicely and should indeed be included:
```suggestion
  - repo: https://github.com/PyCQA/flake8
    rev: v6.1.0
    hooks:
      - id: flake8
  - repo: https://github.com/PyCQA/isort
    rev: v5.12.0
    hooks:
      - id: isort
  - repo: https://github.com/PyCQA/pydocstyle
    rev: 6.3.0
    hooks:
      - id: pydocstyle
```